### ISSUE TITLE
Process substitution in complex command results in file not found

### ISSUE THREAD

**Issue by mparuzel** [msys2/MSYS2-packages#2443]
## Describe the issue

Process substitution works for simple cases, such as `cat <(ls)`, but results in a file not found when used in a more complex command. I'm not sure where the issue lie, whether it's in openssl or the way temporary files are handled.

## Steps to Reproduce the Problem

You do not require anything specific installed other than MSYS2 and OpenSSL.

1. Follow steps 1-4 from here: https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/double-proxy.html
2. When you get to step 4, the process substitution for '-extfile' will cause openssl to not find `/proc/1170/fd/63`.

$ openssl x509 -req -in certs/proxy-postgres-frontend.example.com.csr -CA certs/ca.crt -CAkey certs/ca.key -CAcreateserial -extfile <(printf "subjectAltName=DNS:proxy-postgres-frontend.example.com") -out certs/postgres-frontend.example.com.crt -days 500 -sha256

Can't open /proc/1170/fd/63 for reading, No such file or directory
6716:error:02001003:system library:fopen:No such process:../openssl-1.1.1k/crypto/bio/bss_file.c:69:fopen('/proc/1170/fd/63','r')
6716:error:2006D080:BIO routines:BIO_new_file:no such file:../openssl-1.1.1k/crypto/bio/bss_file.c:76:

3. What is expected is that the command succeeds.

As a work around, the contents of the `printf` can be written to a file and the file name can be provided instead.

## Additional Context: Operating System, Screenshots

- OS: Windows 10 Pro (Version 20H2)
- MSYS2 version 20210228
- OpenSSL version 1.1.1k, 25 Mar 2021

**Comment by 1480c1** [comment]
if you run `echo <(printf "subjectAltName=DNS:proxy-postgres-frontend.example.com")`, what get's printed out?

as a reference, it prints `/dev/fd/63` on my msys2 as `/proc` is linux specific iirc

**Comment by mparuzel** [comment]
```
$ echo <(printf "subjectAltName=DNS:proxy-postgres-frontend.example.com")
/dev/fd/63
```

**Comment by 1480c1** [comment]
Yeah, that's not making much sense either, since I don't think there should be anywhere between msys2 to openssl that would convert `/dev/fd/63` to `/proc/1170/fd/63`

**Comment by 1480c1** [comment]
Just making sure, how did you get your openssl and which openssl package do you have? `msys/openssl` or some other one?

**Comment by mparuzel** [comment]
```
$ pacman -S openssl
warning: openssl-1.1.1.i-1 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (1) openssl-1.1.1.i-1

Total Installed Size:  1.25 MiB
Net Upgrade Size:      0.00 MiB

:: Proceed with installation? [Y/n]
(1/1) checking keys in keyring                                                        [#################################################] 100%
(1/1) checking package integrity                                                      [#################################################] 100%
(1/1) loading package files                                                           [#################################################] 100%
(1/1) checking for file conflicts                                                     [#################################################] 100%
(1/1) checking available disk space                                                   [#################################################] 100%
:: Processing package changes...
(1/1) reinstalling openssl                                                            [#################################################] 100%

$ which openssl
/mingw64/bin/openssl

$ openssl x509 -req -in certs/proxy-postgres-frontend.example.com.csr -CA certs/ca.crt -CAkey certs/ca.key -CAcreateserial -extfile <(printf "subjectAltName=DNS:proxy-postgres-frontend.example.com") -out certs/postgres-frontend.example.com.crt -days 500 -sha256                         Can't open /proc/1306/fd/63 for reading, No such file or directory
9516:error:02001003:system library:fopen:No such process:../openssl-1.1.1k/crypto/bio/bss_file.c:69:fopen('/proc/1306/fd/63','r')
9516:error:2006D080:BIO routines:BIO_new_file:no such file:../openssl-1.1.1k/crypto/bio/bss_file.c:76:
```

**Comment by mparuzel** [comment]
I just realized the openssl being used here is from `/mingw64/bin/`. Version used is `k` while version in MSYS2` is i`.

**Comment by 1480c1** [comment]
So, the `/mingw64/bin/openssl` is telling me you are using the `mingw-w64-x86_64-openssl` package, but I think I might have found the issue
```c
#include <stdio.h>

int main(int argc, char **argv) {
    printf("%s\n", argv[argc - 1]);
}
```
compiling with mingw-w64 gcc results in
```bash
./a.exe <(echo 1)
/proc/1495/fd/63
```

@lazka do you know if this is a cygwin issue?

**Comment by 1480c1** [comment]
As a secondary note @mparuzel, if you want to still use that guide, and you don't need anything mingw-w64 exclusive from openssl, you can try seeing if using msys2.exe instead of mingw64.exe can help you as that one should in theory be more unix path aware and not need the path conversions

as a note with msys gcc
```bash
./a.exe <(echo 1)
/dev/fd/63
```

**Comment by lazka** [comment]
> @lazka do you know if this is an internal path conversion issue?

Based on some testing it seems to skip anything starting in /proc. So I guess this is on purpose somehow
